wallet/models.py

```diff
@@ -35,15 +35,77 @@ class Wallet(models.Model):
         auto_now_add=True, 
     )
 
+    @property
+    def transactions(self):
+        return (self.transactions_outbound.all() | self.transactions_inbound.all())
 
     def get_absolute_url(self):
         return reverse('api_wallet:detail', kwargs={'pk': self.id})
 
+    def deposit(self, value, transaction):
+        """Deposits a value to the wallet. Also updates the transaction
+        balance_dest field
+        """
+        # Save source values on transaction
+        transaction.balance_dest=self.balance + value
+        transaction.save()
+        
+        # Save values on wallet
+        self.balance += value
+        self.save()
+
+    def withdraw(self, value, transaction):
+        """Withdraw's a value from the wallet.
+        Should the withdrawn amount is greater than the
+        balance this wallet currently has, it raises an
+        :mod:`InsufficientBalance` error. This exception
+        inherits from :mod:`django.db.IntegrityError`. So
+        that it automatically rolls-back during a
+        transaction lifecycle.
+        """
+        if value > self.balance:
+            raise InsufficientBalance('This wallet has insufficient balance.')
+        # Save source values on transaction
+        transaction.wallet_source=self
+        transaction.balance_source = self.balance - value
+        transaction.save()
+        
+        # Save values on wallet
+        self.balance -= value
+        self.save()
+
+    def transfer(self, wallet, value):
+        """Transfers an value to another wallet.
+        Uses `deposit` and `withdraw` internally.
+        Creates the transaction record and links it to
+        the 2 wallets (source and destination wallets)
+        """
+        transaction = Transaction.objects.create(
+            value=value,
+            wallet_dest=wallet,
+            wallet_source=self,
+            balance_dest=0,
+            balance_source=0,
+        )
+        transaction.save()
+        self.withdraw(value, transaction)
+        wallet.deposit(value, transaction)
```

Comment:
There is four DB accesses when just for creating a single transaction. 

One option... is to remove every _save()_ from _withdraw_ and _deposit_ functions. And I think it'll still working the exact same way.